File: eRestorante/eRestorante.Services/Database/Reservation.cs
using System;
using System.Collections.Generic;

namespace eRestorante.Services.Database;

public partial class Reservation
{
    public int ReservationId { get; set; }

    public DateOnly ReservationDate { get; set; }

    public TimeOnly ReservationTime { get; set; }

    public int NumberOfGuests { get; set; }

    public int NumberOfHours { get; set; }

    public int ReservationStatus { get; set; }

    public int? CustomerId { get; set; }

    public string? ReservationReason { get; set; }

    public int? NumberOfMinors { get; set; }

    public string ContactPhone { get; set; } = null!;

    public string SpecialWishes { get; set; } = null!;

    public virtual Customer? Customer { get; set; }
}

File: eRestorante/eRestoranteUI/erestorante_mobile/lib/models/reservation.dart
import 'package:erestorante_mobile/models/customer.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable(explicitToJson: true)
class Reservation{
  int? reservationId;
  String? reservationDate;
  String? reservationTime;
  int? numberOfGuests;
  int? numberOfHours;
  int? reservationStatus;
  int? customerId;
  String? reservationReason;
  int? numberOfMinors;
  String? contactPhone;
  String? specialWishes;
  Customer? customer;

  Reservation(this.reservationId, this.reservationDate, this.reservationTime, this.numberOfGuests, this.numberOfHours, this.reservationStatus, this.customerId, this.reservationReason, this.numberOfMinors, this.contactPhone, this.specialWishes, this.customer);
  factory Reservation.fromJson(Map<String, dynamic> json) => _$ReservationFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}